challenge_help_me_carry/src/challenge_help_me_carry.py

```diff
@@ -0,0 +1,171 @@
+print "=============================================="
+
+class WaitForOperatorCommand(smach.State):
+    def __init__(self, robot, command_options, commands_as_outcomes=False):
+
+        if commands_as_outcomes:
+            # each possible command is a separate outcome
+            smach.State.__init__(self, outcomes=command_options + ['abort'])
+        else:
+            # outcome is success or abort, recognized command is returned using output_keys
+            smach.State.__init__(self, outcomes=['success', 'abort'],  output_keys=['command_recognized'])
+
+        self._robot = robot
+        self._command_options = command_options
+
+    def _listen_for_commands(self, tries=3, time_out = rospy.Duration(30)):
+        for i in range(0, tries):
+            result = self._robot.ears.recognize("<option>", {"option" : self._command_options}, time_out)
+            command_recognized = result.result
+
+            if command_recognized == "":
+                self._robot.speech.speak("I am still waiting for a command and did not hear anything")
+
+            elif command_recognized in self._command_options:
+                self._robot.speech.speak(command_recognized + " OK")
+                return "success", command_recognized
+
+            else:
+                self._robot.speech.speak("I don't understand, I expected a command like " + self._command_options)
+
+        self._robot.speech.speak("I did not recognize a command and will stop now")
+        return "abort", ""
+
+    def execute(self, userdata):
+        # Stop the base
+        self._robot.base.local_planner.cancelCurrentPlan()
+        self._robot.head.look_at_standing_person()
+
+        outcome, userdata.command_recognized = self._listen_for_commands()
+
+        return outcome
+
+class StoreCarWaypoint(smach.State):
+    def __init__(self, robot):
+        smach.State.__init__(self, outcomes=['success', 'abort'])
+        self._robot = robot
+        robot.base.local_planner.cancelCurrentPlan()
+
+    def execute(self, userdata):
+        success = self._robot.ed.update_entity(id=challenge_knowledge.waypoints.car, posestamped=self._robot.base.get_location(), type="waypoint")
+
+        if success:
+            return "success"
+        else:
+            return "abort"
+
+class NavigateToRoom(smach.State):
+    def __init__(self, robot):
+        smach.State.__init__(self,
+                             outcomes=['success', 'abort'],
+                             input_keys=['target_room'])
+        self._robot = robot
+        robot.base.local_planner.cancelCurrentPlan()
+
+    def execute(self, userdata):
+
+        target_waypoint, target_radius = challenge_knowledge.commands.waypoint_for_room[userdata.target_room]
+
+        navigateToWaypoint = states.NavigateToWaypoint(self._robot, EntityByIdDesignator(self._robot, id=target_waypoint), target_radius)
+
+        return navigateToWaypoint.execute()
+
+
+def setup_statemachine(robot):
```

Comment:
Please take a look at some of the other challenges. The main entry point should also be a class inheriting from smach.Statemachine